src/components/App/index.tsx

```diff
@@ -0,0 +1,294 @@
+    () => [
+      { term, versionName, oscar, ...filteredScheduleData },
+      { setTerm, setVersionName, setOscar, patchScheduleData }
+    ],
+    [
+      term,
+      versionName,
+      oscar,
+      filteredScheduleData,
+      setTerm,
+      setVersionName,
+      setOscar,
+      patchScheduleData
+    ]
+  );
+  const versionsContextValue = useMemo<VersionsContextValue>(
+    () => [{ ...versionLists }, { patchVersionsData }],
+    [versionLists, patchVersionsData]
+  );
+
+  // display popup when first visiting the site
+  useEffect(() => {
+    const cookieKey = 'visited-merge-notice';
+    if (!Cookies.get(cookieKey)) {
+      swal({
+        button: 'Got It!',
+        content: (
+          <div>
+            <img
+              style={{ width: '175px' }}
+              alt="GT Scheduler Logo"
+              src="/mascot.png"
+            />
+            <h1>GT Scheduler</h1>
+            <p>
+              Hey there, yellow jackets!{' '}
+              <a href="https://github.com/gt-scheduler">GT Scheduler</a> is a
+              new collaboration between{' '}
+              <a href="https://bitsofgood.org/">Bits of Good</a> and{' '}
+              <a href="https://jasonpark.me/">Jason (Jinseo) Park</a> aimed at
+              making class registration easier for everybody! Now, you can
+              access course prerequisites, instructor GPAs, live seating
+              information, and more all in one location.
+            </p>
+            <p>
+              If you enjoy our work and are interested in contributing, feel
+              free to{' '}
+              <a href="https://github.com/gt-scheduler/website/pulls">
+                open a pull request
+              </a>{' '}
+              with your improvements. Thank you and enjoy!
+            </p>
+          </div>
+        )
+      });
+
+      Cookies.set(cookieKey, 'true', { expires: 365 });
+    }
+  }, []);
+
+  // Fetch the current term's scraper information
+  useEffect(() => {
+    setOscar(null);
+    if (term) {
+      axios
+        .get(`https://gt-scheduler.github.io/crawler/${term}.json`)
+        .then((res) => {
+          const newOscar = new Oscar(res.data);
+          setOscar(newOscar);
+        });
+    }
+  }, [term]);
+
+  // Fetch all terms via the GitHub API
+  useEffect(() => {
+    axios
+      .get(
+        'https://api.github.com/repos/gt-scheduler/crawler/contents?ref=gh-pages'
+      )
+      .then((res) => {
+        const newTerms = (res.data as { name: string }[])
+          .map((content) => content.name)
+          .filter((name: string) => /\d{6}\.json/.test(name))
+          .map((name: string) => name.replace(/\.json$/, ''))
+          .sort()
+          .reverse();
+        setTerms(newTerms);
+      });
+  }, [setTerms]);
+
+  // Set the term to be the first one if it is unset
+  // (once the terms load)
+  useEffect(() => {
+    const t = Cookies.get('term');
+    if (!term && (!t || t === 'undefined')) {
```

Comment:
What is the point of the up-to-date value `t` here, and in general, why were the changes needed? Additionally, a more descriptive variable name would be appropriate.

Reply:
This "t" variable reads the term value from cookie instead of relying on the "term" state. When app first loads and reaches this useEffect(), the "term" state is always undefined; if you check the current implementation on gt-scheduler.org, no matter which term you were at when you closed the page, you always go back to Fall 2021 when you reopen the app. This is caused by the issue I mentioned, and my code will read from cookie and go back to the same term the user left at. I'll rename "t" to "term_from_cookie" for clarity.

Reply:
That makes sense, thanks